src/ophys_etl/transforms/sine_dewarp.py

```diff
@@ -0,0 +1,606 @@
+
+    Parameters
+    ----------
+    movie: np.ndarray
+        The movie that is being dewarped.
+    aL: float
+        Information about the rig used for the experiment.
+    aR: float
+        Information about the rig used for the experiment.
+    bL: float
+        Information about the rig used for the experiment.
+    bR: float
+        Information about the rig used for the experiment.
+    noise_reduction: int
+        The noise reduction method that will be used in dewarping.
+    Returns
+    -------
+    dict
+        Various computed information about the video.
+    """
+    # assume flat area aL - (512-aR)
+
+    meanimg = np.mean(movie, axis=0)  # use avgimg to compute mode
+    meanimg = meanimg/256          # less noisy to get mode from 8bit histogram
+    meanimg = meanimg.astype(int)  # less noisy to get mode from 8bit histogram
+
+    # full movie got Memory Error
+    stdvimg = np.std(movie[::8], axis=0)  # use avgimg to compute mode
+    stdvimg = stdvimg/256          # less noisy to get mode from 8bit histogram
+    stdvimg = stdvimg.astype(int)  # less noisy to get mode from 8bit histogram
+
+    # Left side
+    xindexL, xindexLB = get_xindex(aL, bL)
+
+    # Right side
+    xindexR, xindexRB = get_xindex(aR, bR)
+
+    modelist, mean_maxcount = mode(meanimg)
+    mean_modevalue = modelist[0]
+    mean_minvalue = np.min(meanimg)
+
+    # IMPORTANT!: estmated modevalue is from 8bit img, convert back to 16bit
+    mean_modevalue = mean_modevalue * 256
+    mean_minvalue = mean_minvalue * 256
+
+    modelist, stdv_maxcount = mode(stdvimg)
+    stdv_modevalue = modelist[0]
+    stdv_minvalue = np.min(stdvimg)
+
+    # IMPORTANT!: estmated modevalue is from 8bit img, convert back to 16bit
+    stdv_modevalue = stdv_modevalue * 256
+    stdv_minvalue = stdv_minvalue * 256
+
+    j = 0
+    while xindexLB[j] < 0.0 or xindexL[j] < 0:
+        j = j + 1
+    else:
+        bgfactorL = xindexLB[j+1] - xindexLB[j]
+        L = j
+
+    j = 0
+    while xindexRB[j] < 0.0 or xindexR[j] < 0:
+        j = j + 1
+    else:
+        bgfactorR = xindexRB[j+1] - xindexRB[j]
+        R = j
+
+    # TODO: Check on this. Make sure it wasn't an
+    # error in the original code
+    if 0 != noise_reduction:
+        bgfactorL = 1
+        bgfactorR = 1
+
+    table = {
+        "mean_modevalue": mean_modevalue,
+        'mean_maxcount': mean_maxcount,
+        "mean_minvalue": mean_minvalue,
+        "stdv_modevalue": stdv_modevalue,
+        "stdv_maxcount": stdv_maxcount,
+        "stdv_minvalue": stdv_minvalue,
+        "bgfactorL": bgfactorL,
+        "bgfactorR": bgfactorR,
+        "L": L,
+        "R": R,
+        "xindexL": xindexL,
+        "xindexLB": xindexLB,
+        "xindexR": xindexR,
+        "xindexRB": xindexRB,
+        "aL": aL,
+        "aR": aR,
+        "bL": bL,
+        "bR": bR
+    }
+    logging.debug(table)
+
+    return table
+
+
+def parse_input(data: json):
```

Comment:
to be consistent with most of the rest of this repo, it would be nice to eventually replace this parsing with an argschema

Reply:
please add to the next ticket